core/auth/fake/identifier_test.go

```diff
@@ -0,0 +1,304 @@
+package fake
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"flamingo.me/flamingo/v3/framework/flamingo"
+
+	"flamingo.me/flamingo/v3/framework/web"
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	mockRouter struct {
+		broker string
+	}
+)
+
+var _ web.ReverseRouter = (*mockRouter)(nil)
+
+// Relative mock action
+func (m *mockRouter) Relative(_ string, _ map[string]string) (*url.URL, error) {
+	panic("not implemented")
+}
+
+// Absolute mock action
+func (m *mockRouter) Absolute(_ *web.Request, _ string, _ map[string]string) (*url.URL, error) {
+	resultURL := &url.URL{}
+
+	return resultURL.Parse(strings.ReplaceAll("/core/auth/login/:broker", ":broker", m.broker))
+}
+
+func Test_Identifier_Authenticate(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		config fakeConfig
+	}
+
+	type args struct {
+		r *web.Request
+	}
+
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   web.Result
+	}{
+		{
+			name: "simple identifier call with empty template to test fallback template",
+			fields: fields{
+				config: fakeConfig{},
+			},
+			args: args{
+				r: addRequestParameters(web.CreateRequest(
```

Comment:
Please create the request in a helper method. This code is duplicated 7 times just in this test.